code/Forms/InternalLinkFormFactory.php

```diff
@@ -0,0 +1,40 @@
+<?php
```

Comment:
admin doesn't know anything about page type (the internal link); This code is all CMS specific.